enumerable_project.rb

```diff
@@ -0,0 +1,139 @@
+ Copyright (c) Muhammasd Arslan
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each do |x|
+      arr << x if yield(x) == true
+    end
+    arr
+  end
+
+  def my_all?
+    return self unless block_given?
+
+    my_each do |x|
+      return false if yield(x) == false
+    end
+    true
+  end
+
+  def my_any?
+    return self unless block_given?
+
+    my_each do |x|
+      return true if yield(x) == true
+    end
+    false
+  end
+
+  def my_none?
+    return self unless block_given?
+
+    my_each do |x|
+      return false if yield(x) == true
+    end
+    true
+  end
```

Comment:
my_none?:

When no block or argument is given returns true only if none of the collection members is true
`p [nil, false, true, []].my_none?` should return false

When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`p ['hello'].my_none?(5)`  should return true

when a class is passed as an argument returns true if none of the collection is a member of such class
`p [3,3,3,3].my_none?(String)` should return true

when a Regex is passed as an argument returns true only if none of the collection matches the Regex
`p ['hello','milk','milo'].my_none?(/z/)` should return true